docs/spec/spec.md

```diff
@@ -0,0 +1,322 @@
+        importParam3: 123.45,
+        importParam4: 123.456
+    };
+
+    ExportParams? result = check jcoClient->execute("TEST_FUNCTION", importParams);
+    if result is jco:ExportParams {
+        io:println("Result: ", result);
+    } else {
+        io:println("Error: Function execution failed");
+    }
+}
+```
+
+##### 2.1.2.2 Send IDocs to an SAP System
+
+The RFC Client also supports sending IDocs to an SAP system, allowing you to automate the exchange of structured data.
+
+An IDoc can be sent using the following function:
+
+```ballerina
+# Send the iDoc
+#
+# + iDoc - The XML string of the iDoc
+# + iDocType - The type of the iDoc
+# + return - An error if the execution fails
+remote function sendIDoc(xml iDoc, IDocType iDocType = jco:DEFAULT) returns Error
+```
+
+The input parameters require `iDoc`, with an optional `iDocType`. The `iDoc` represents the content of the IDoc in XML format. The `iDocType` specifies the version of the IDoc being sent and can be set to `DEFAULT`, `VERSION_2`, `VERSION_3`, `VERSION_3_IN_QUEUE`, or `VERSION_3_IN_QUEUE_VIA_QRFC`. If no `iDocType` is provided, the `DEFAULT` type will be applied.
+
+Users can invoke it as shown below:
+
+```ballerina
+public function main() returns error? {
+    xml iDoc = check io:fileReadXml("resources/sample.xml");
+    check jcoClient->sendIDoc(iDoc);
+    io:println("IDoc sent successfully.");
+}
+```
+
+#### 2.1.3 Data Type Mapping Between Ballerina and SAP JCo
+
+The following table maps Ballerina data types to their corresponding SAP JCo types:
+
+| Ballerina Type                        | SAP JCo Type   | SAP Data Type         |
+| ------------------------------------- | -------------- | --------------------- |
+| string                                | char           | CHAR                  |
+| string                                | char[]         | STRING, SSTRING       |
+| time:Date                             | Date           | DATE                  |
+| time:TimeOfDay                        | Time           | TIME                  |
+| int                                   | short          | INT1                  |
+| int                                   | int            | INT2                  |
+| int                                   | long           | INT4                  |
+| float                                 | float          | FLOAT                 |
+| decimal                               | double         | DEC, QUAN, CURR, PREC |
+| byte                                  | byte           | INT1                  |
+| byte[]                                | byte[]         | RAW, LRAW, BYTE       |
+| decimal                               | BigDecimal     | DEC, QUAN, CURR       |
+| int                                   | BigInteger     | NUM                   |
+| record (with/without rest fields)     | JCoStructure   | STRUCTURE             |
+| record[] (with/without rest fields)   | JCoTable       | TABLE                 |
+
+### 2.2 Listener
```

Comment:
We have not included the functions defined in the `jco:Listener`. Shall we add those too ?

Reply:
Changed with [cf4c263](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/56/commits/cf4c263ee0ab616f3fac2aa71fcabf05d0712c17).